Tensile/Source/client/main.cpp

```diff
@@ -523,19 +580,29 @@ int main(int argc, const char* argv[])
     size_t maxWorkspaceSize
         = getMaxWorkspace(library, hardware, args, problems, firstProblemIdx, lastProblemIdx);
     maxWorkspaceSize = std::min(maxWorkspaceSize, maxWorkspaceSizeLimit);
+    size_t flush_count = args["flush-count"].as<size_t>();
+    size_t flush_memory_size = args["flush-memory-size"].as<size_t>();
 
-    auto dataInit = DataInitialization::Get(args, problemFactory, maxWorkspaceSize);
-
+    std::vector<std::shared_ptr<DataInitialization>> dataInit;
     auto solutionIterator = SolutionIterator::Default(library, hardware, args);
 
     MetaRunListener listeners;
 
-    listeners.addListener(dataInit);
+    flush_count = calculate_flush_count(flush_count, 
+                                            flush_memory_size,
+                                            problemFactory);
+
+    for(size_t i = 0; i<flush_count; i++)
+    {
+        dataInit.push_back(DataInitialization::Get(args, problemFactory, maxWorkspaceSize));
+        listeners.addListener(dataInit[i]);
+    }
+
     listeners.addListener(solutionIterator);
     listeners.addListener(std::make_shared<ProgressListener>(args));
     if(runKernels)
     {
-        listeners.addListener(std::make_shared<ReferenceValidator>(args, dataInit));
+        listeners.addListener(std::make_shared<ReferenceValidator>(args, dataInit[0]));
```

Comment:
Adding only single listener to accommodate validation of the first warmup as previously done